subgraphs/maple-finance/src/common/mappingHelpers/update/intervalUpdate.ts

```diff
@@ -131,19 +131,28 @@ function intervalUpdateMarket(event: ethereum.Event, market: Market): Market {
 
 function intervalUpdateProtocol(event: ethereum.Event, marketBefore: Market, marketAfter: Market): void {
     const protocol = getOrCreateProtocol();
+    const marketIDs = protocol._marketIDs;
 
-    protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.plus(
-        marketAfter.totalValueLockedUSD.minus(marketBefore.totalValueLockedUSD)
-    );
+    let totalValueLockedUSD = ZERO_BD;
+    let totalDepositBalanceUSD = ZERO_BD;
+    let totalBorrowBalanceUSD = ZERO_BD;
 
-    protocol.totalDepositBalanceUSD = protocol.totalDepositBalanceUSD.plus(
-        marketAfter.totalDepositBalanceUSD.minus(marketBefore.totalDepositBalanceUSD)
-    );
+    for (let i = 0; i < marketIDs.length; i++) {
+        const market = Market.load(marketIDs[i]);
+        if (!market) {
+            // fail safe in case market does not exist for some reason
+            continue;
+        }
```

Comment:
I always have this internal debate on what we should do in these cases. This should not happen at all, unless there is a bug when creating the market, or some other function adds stuff to `marketIDs` for some reason.

Do we want it to fail silently? I would say to at least log an error (stopping the execution feels too extreme), but since our logs are lost in space it, it's not like it will make a difference 😅

Reply:
log added